tensorflow/cc/framework/gradients.cc

```diff
@@ -77,6 +77,10 @@ class SymbolicGradientBuilder {
   Status CallGradFunction(const Operation& op,
                           const std::vector<Output>& grad_inputs,
                           std::vector<Output>* grad_outputs);
+  
+  // If a node is reachable when doing a BFS from outputs toward inputs
+  // its id used as index in reachable_nodes is set to true.
```

Comment:
How about: "Returns a list mapping whether each node in the graph is reachable from outputs_. Keyed by node id."